### DISCUSSION TITLE
Getting ModuleNotFoundError when trying to run cs_tools from PS on Windows 10

### DISCUSSION THREAD

**Discussion by bbiggs-gaig** [thoughtspot/cs_tools#104] [Q&A] [answered]
After installing Python and cs_tools, I am getting the error below with I try to run cs_tools. Thoughts on what I might be doing wrong?

```
PS C:\Users\User> cs_tools --version
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\User\AppData\Local\Programs\Python\Python311\Scripts\cs_tools.exe\__main__.py", line 4, in <module>
  File "C:\Users\User\AppData\Local\Programs\Python\Python311\Lib\site-packages\cs_tools\cli\main.py", line 15, in <module>
    import typer
ModuleNotFoundError: No module named 'typer'
``` 

Here is the version of Python I have installed:
```
PS C:\Users\User> python
Python 3.11.8 (tags/v3.11.8:db85d51, Feb  6 2024, 22:03:32) [MSC v.1937 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
``` 
Thanks!

**Comment by boonhapus** [comment]
Hey @bbiggs-gaig , how did you go about installing `cs_tools` ? It looks like you've likely only installed the library code itself. If you're trying to get access to the command line, you can run the installer code.

```powershell
(Invoke-WebRequest `
    -Uri https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py `
    -UseBasicParsing `
).Content | python - --reinstall
```

**Reply by bbiggs-gaig** [reply, marked as answer]
Thanks for the reply! I ran the command shown on the [first-time-install](https://thoughtspot.github.io/cs_tools/tutorial/install/#first-time-install) page. Twice. Didn't know about the `--reinstall` option.

I ended up working with ThoughtSpot Support, and I had to run the following commands from PowerShell to get additional Python modules installed. I'm not sure why they didn't get installed when I installed Python and CS Tools initially. ❓❓❓

```
pip install typer
pip install uvicorn
pip install fastapi
``` 

After I ran these, cs_tools is working fine from the command line.

**Reply by boonhapus** [reply]
Interesting, thanks for the info @bbiggs-gaig . I'm not sure why they didn't get installed either! I'm currently running py 3.11.8 on my production Windows server as well, odd. 😅 

In the future, myself or the other folks from the CS Tools team are happy work live with you to get things resolved! We have a thoughtspot-internal slack channel where all the Issues and Discussions get posted to help us keep track. The ThoughtSpot Support team is primarily meant to help with cluster-level issues and while they're familiar with this project, it's probably simpler for our group to handle it.

**Reply by bbiggs-gaig** [reply]
Good to know. I guess it wasn't _technically_ ThoughtSpot Support. It was the Migration team we are working with to move from on-prem to the cloud. Thanks!
